testing/go/record_test.go

```diff
@@ -0,0 +1,364 @@
+// Copyright 2025 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package _go
+
+import (
+	"testing"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+func TestRecords(t *testing.T) {
+	RunScripts(t, []ScriptTest{
+		{
+			Name: "Record cannot be used as column type",
+			SetUpScript: []string{
+				"CREATE TABLE t2 (pk INT PRIMARY KEY, c1 VARCHAR(100));",
+			},
+			Assertions: []ScriptTestAssertion{
+				{
+					Query:       "CREATE TABLE t (pk INT PRIMARY KEY, r RECORD);",
+					ExpectedErr: `column "r" has pseudo-type record`,
+				},
+				{
+					Query:       "ALTER TABLE t2 ADD COLUMN c2 RECORD;",
+					ExpectedErr: `column "c2" has pseudo-type record`,
+				},
+				{
+					Query:       "ALTER TABLE t2 ALTER COLUMN c1 TYPE RECORD;",
+					ExpectedErr: `column "c1" has pseudo-type record`,
+				},
+				{
+					Query:       "CREATE DOMAIN my_domain AS record;",
+					ExpectedErr: `"record" is not a valid base type for a domain`,
+				},
+				{
+					Query:       "CREATE SEQUENCE my_seq AS record;",
+					ExpectedErr: "sequence type must be smallint, integer, or bigint",
+				},
+				{
+					Query:       "CREATE TYPE outer_type AS (id int, payload record);",
+					ExpectedErr: `column "payload" has pseudo-type record`,
+				},
+			},
+		},
+		{
+			Name: "Casting to record",
+			Assertions: []ScriptTestAssertion{
+				{
+					Query:    "select row(1, 1)::record;",
+					Expected: []sql.Row{{"(1,1)"}},
+				},
+			},
+		},
+		{
+			// TODO: Wrapping table rows with ROW() is not supported yet. Planbuilder assumes the
+			//       table alias is a column name and not a table.
+			Skip: true,
```

Comment:
Perhaps we could skip the individual assertions rather than the test at large. I had to double take when I saw the `TODO: ERROR`, but didn't see any skips until I saw the skip was on the test itself.